py/dml/ctree.py

```diff
@@ -4386,7 +4583,25 @@ def mkSubRef(site, expr, sub, op):
         else:
             raise EREF(site, sub, expr)
 
-    elif isinstance(expr, AddressOf) and op == '->':
+    if isinstance(expr, NonValue):
+        if op == '.':
```

Comment:
Untested corner:
```
some_template t = cast(obj, some_template);
some_template *tp = &t;
/// ERROR ENVAL (?)
tp->templates.parent_template.m
```

Reply:
What part of that is untested, specifically? That you can do `tp->templates` is tested in T_tqmic, and that a resolved TQMI ref gives ENVAL I don't feel needs to be tested. As stated elsewhere it's just a `NodeRef`/`TraitMethodDirect`.

Reply:
Sorry for the noise! I misunderstood the code and thought it would give an error, but didn't see a matching error test. Indeed there is a positive test for it.